scipy/stats/mstats_basic.py

```diff
@@ -356,39 +356,171 @@ def msign(x):
 
 
 def pearsonr(x, y):
-    """
-    Calculates a Pearson correlation coefficient and the p-value for testing
-    non-correlation.
+    r"""
+    Pearson correlation coefficient and p-value for testing non-correlation.
 
-    The Pearson correlation coefficient measures the linear relationship
-    between two datasets. Strictly speaking, Pearson's correlation requires
-    that each dataset be normally distributed. Like other correlation
+    The Pearson correlation coefficient [1]_ measures the linear relationship
+    between two datasets.  The calculation of the p-value relies on the
+    assumption that each dataset is normally distributed.  (See Kowalski [3]_
+    for a discussion of the effects of non-normality of the input on the
+    distribution of the correlation coefficient.)  Like other correlation
     coefficients, this one varies between -1 and +1 with 0 implying no
-    correlation. Correlations of -1 or +1 imply an exact linear
-    relationship. Positive correlations imply that as `x` increases, so does
-    `y`. Negative correlations imply that as `x` increases, `y` decreases.
+    correlation. Correlations of -1 or +1 imply an exact linear relationship.
+    Positive correlations imply that as x increases, so does y. Negative
+    correlations imply that as x increases, y decreases.
 
     The p-value roughly indicates the probability of an uncorrelated system
     producing datasets that have a Pearson correlation at least as extreme
-    as the one computed from these datasets. The p-values are not entirely
-    reliable but are probably reasonable for datasets larger than 500 or so.
+    as the one computed from these datasets.
 
     Parameters
     ----------
-    x : 1-D array_like
-        Input
-    y : 1-D array_like
-        Input
+    x : (N,) array_like
+        Input array.
+    y : (N,) array_like
+        Input array.
 
     Returns
     -------
-    pearsonr : float
-        Pearson's correlation coefficient, 2-tailed p-value.
+    r : float
+        Pearson's correlation coefficient.
+    p-value : float
+        Two-tailed p-value.
+
+    Warns
+    -----
+    PearsonRConstantInputWarning
+        Raised if an input is a constant array.  The correlation coefficient
+        is not defined in this case, so ``np.nan`` is returned.
+
+    PearsonRNearConstantInputWarning
+        Raised if an input is "nearly" constant.  The array ``x`` is considered
+        nearly constant if ``norm(x - mean(x)) < 1e-13 * abs(mean(x))``.
+        Numerical errors in the calculation ``x - mean(x)`` in this case might
+        result in an inaccurate calculation of r.
```

Comment:
I'm not sure what the correct formatting would be here. On one hand, _x_ is a parameter, so numpydoc would want single backticks, this would be rendered italicized. In the expression, though, `x` is certainly "code", so there would be double-backticks around that and it will render in monospaced font. But then, `x` will either look different than the parameter it represents or it will look different from the same symbol used in this code snippet. Since you're caught between a rock and hard place, I guess this is fine.

I'm also not certain what to do about "r". numpydoc says "Enclose variables in single backticks." in the Parameters section. Does that extend to returns?

This is why I'd suggest we eliminate the confusion both for readers and writers by just making all variables/parameters/returns/(and anything else that represents code) also appear in monospaced font. Do you agree?

Reply:
see https://github.com/scipy/scipy/pull/14183#discussion_r646905768